### ISSUE TITLE
1D Transposed Convolutional Layer

### ISSUE THREAD

**Issue by lars-reimann** [Safe-DS/Library#854]
### Is your feature request related to a problem?

We only have 1D transposed convolutions so far. 1D transposed convolutions can also be useful, e.g. for time series.

### Desired solution

* Add a new class `safeds.ml.nn.layers.ConvolutionalTranspose1DLayer`.
* Superclass: `Convolutional1DLayer` (see #853)
* Constructor parameters: 
  * `output_channel: int`
  * `kernel_size: int`
  * **keyword-only** `stride: int = 1`
  * **keyword-only** `padding: int = 0`
  * **keyword-only** `output_padding: int = 0`
  * Call `_check_bounds` to ensure that:
    * `output_channel` >= 1
    * `kernel_size` >= 1
    * `stride` >= 1
    * `padding` >= 0
    * `output_padding` >= 0
* Implementation:
  * Use PyTorch's [`ConvTranspose1D`](https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose1d.html) internally
